Backpack/src/main/java/net/skyscanner/backpack/badge/BpkBadge.kt

```diff
@@ -156,6 +191,14 @@ open class BpkBadge @JvmOverloads constructor(
             setBackground(bgColor)
         }
         this.gravity = Gravity.CENTER
+
+        compoundDrawablePadding = iconPadding
+        icon?.setTint(context.getColor(type.iconColor))
+    }
+
+    private fun updateIcon() {
+        setCompoundDrawablesRelative(icon, null, null, null)
```

Comment:
i think you need to move setting the tint here as otherwise icons set programmatically won't be tinted correctly